src/components/Figure/Image.js

```diff
@@ -26,29 +53,44 @@ class Image extends Component {
       maxWidth,
       size: sizeProp,
       aboveTheFold,
-      enableGallery = true
+      enableGallery = true,
+      gallerySize
     } = this.props
 
     const onClick = (enableGallery && this.context.toggleGallery) ? this.context.toggleGallery : () => {}
 
     const size = sizeProp || (sizeProp === undefined && imageSizeInfo(src))
     const aspectRatio = size ? size.width / size.height : undefined
 
+    const handleClick = () => onClick(src)
+
     const image = isFinite(aspectRatio)
       ? <LazyImage attributes={attributes} visible={aboveTheFold}
           aspectRatio={aspectRatio}
-          src={src} srcSet={srcSet} alt={alt} onClick={() => onClick(src)} />
+          src={src} srcSet={srcSet} alt={alt} onClick={handleClick} />
       : <img {...attributes} {...styles.image}
-          src={src} srcSet={srcSet} alt={alt} onClick={() => onClick(src)} />
+          src={src} srcSet={srcSet} alt={alt} onClick={handleClick} />
+
+    let wrappedImage = image
 
     if (maxWidth) {
-      return (
+      wrappedImage = 
         <span {...styles.maxWidth} style={{maxWidth}}>
-          {image}
+          {wrappedImage}
         </span>
-      )
     }
-    return image
+    if (src && gallerySize > 0) {
+      wrappedImage =
+        <>
+          {wrappedImage}
+          <GalleryButton gallerySize={gallerySize} onClick={handleClick} />
+        </>
+    }
+    return (
+      <div {...styles.imageContainer}>
+        {wrappedImage}
```

Comment:
```suggestion
        {wrappedImage}
        {gallerySize > 0 && <GalleryButton gallerySize={gallerySize} onClick={handleClick} />}
```
and rm the `if (src && gallerySize > 0)` above